newton/_src/sim/articulation.py

```diff
@@ -500,6 +529,9 @@ def reconstruct_angular_q_qd(q_pc: wp.quat, w_err: wp.vec3, X_wp: wp.transform,
 
 @wp.kernel
 def eval_articulation_ik(
+    articulation_start: wp.array(dtype=int),
+    articulation_count: int,  # total number of articulations
+    articulation_indices: wp.array(dtype=int),  # can be None, articulation indices to process
```

Comment:
Can we also have `articulation_mask` here to match the eval_fk function? This can be exposed as a optional parameter in `eval_ik` then as well.